pkg/apis/servicecatalog/v1beta1/conversion.go

```diff
@@ -61,3 +68,74 @@ func ServiceInstanceFieldLabelConversionFunc(label, value string) (string, strin
 		return "", "", fmt.Errorf("field label not supported: %s", label)
 	}
 }
+
+// ConvertClusterServiceClassToProperties takes a Service Class and pulls out the
+// properties we support for filtering, converting them into a map in the
+// expected format.
+func ConvertClusterServiceClassToProperties(serviceClass *ClusterServiceClass) filter.Properties {
```

Comment:
this whole class is hand written... I am not aware of a generator that flattens objects. Is there one? We might not want to use it because we are only wanting to look at a couple fields